packages/backend/src/utils/npm.ts

```diff
@@ -0,0 +1,127 @@
+import { exec, execSync } from "child_process";
+import { promisify } from "util";
+import { platform } from "os";
+import * as _ from "lodash";
+import * as customLocation from "./customLocation";
+import * as sudo from "sudo-prompt";
+import { promises, constants } from "fs";
+import messages from "../messages";
+import { vscode } from "./vscodeProxy";
+import * as path from "path";
+
+export const isWin32 = platform() === "win32";
+const NPM = isWin32 ? "npm.cmd" : "npm";
+
+const NPM_REGISTRY_HOST = _.get(process, "env.NPM_CFG_REGISTRY", "http://registry.npmjs.com/");
+const SEARCH_QUERY_PREFIX = `${NPM_REGISTRY_HOST}-/v1/search?text=`;
+const SEARCH_QUERY_SUFFIX = "keywords:yeoman-generator &size=25&ranking=popularity";
+
+const CANCELED = "Action cancelled";
+const HAS_ACCESS = "Has Access";
+
+const enum choices {
+  set_defaul_location,
+  change_global_owner,
+  cancelled,
+}
+
+class Command {
+  private readonly globalNodeModulesPath;
+  private readonly globalPath;
+  private readonly CHANGE_OWNER_FOR_GLOBAL;
+  private readonly SET_DEFAULT_LOCATION;
+
+  constructor() {
+    this.globalNodeModulesPath = _.trim(execSync(`${NPM} root -g`).toString());
+
+    const nmLength = path.join(path.sep, "node_modules").length;
+    this.globalPath = this.globalNodeModulesPath.substring(0, this.globalNodeModulesPath.length - nmLength);
+
+    this.CHANGE_OWNER_FOR_GLOBAL = messages.change_owner_for_global(this.globalPath);
+    this.SET_DEFAULT_LOCATION = messages.set_default_location(customLocation.DEFAULT_LOCATION);
+  }
+
+  private getGenLocationParams(): string {
+    const customInstallationPath = customLocation.getPath();
+    return _.isEmpty(customInstallationPath) ? "-g" : `--prefix ${customInstallationPath}`;
+  }
+
+  public getGlobalNodeModulesPath(): string {
+    return this.globalNodeModulesPath;
+  }
+
+  public async execCommand(arg: string): Promise<any> {
+    return promisify(exec)(arg);
+  }
+
+  public getGensQueryURL(query: string, recommended: string): string {
+    return encodeURI(`${SEARCH_QUERY_PREFIX} ${query} ${recommended} ${SEARCH_QUERY_SUFFIX}`);
+  }
+
+  private async _install(genName: string): Promise<any> {
+    const locationParams = this.getGenLocationParams();
+    const command = `${NPM} install ${locationParams} ${genName}@latest`;
+    return this.execCommand(command);
+  }
+
+  public async uninstall(genName: string): Promise<any> {
+    const locationParams = this.getGenLocationParams();
+    const command = `${NPM} uninstall ${locationParams} ${genName}`;
+    return this.execCommand(command);
+  }
+
+  private async sudoExec(command: string) {
+    return new Promise((resolve, reject) => {
+      sudo.exec(command, {}, (err, script) => {
```

Comment:
```suggestion
      sudo.exec(command, {name: 'Application Wizard'}, (err, script) => {
```

Need to pass a name to the options param

Reply:
on Windows options.name is not supported

Reply:
I fixed it